### ISSUE TITLE
[firebase_messaging] getToken() did not complete or return when called after deleteInstanceID() in iOS

### ISSUE THREAD

**Issue by skybur** [FirebaseExtended/flutterfire#258]
**Describe the bug**
`getToken()` did not complete/return when being called after `deleteInstanceID()` in iOS

**To Reproduce**
1. Request a Token using `getToken()`
2. Token will be returned if it's requested the first time 
3. Call `deleteInstanceID()`
4. Request a Token again using `getToken()`
5. `getToken()` Future did not complete 

**Expected behavior**
`getToken()` will complete and returns a new Token

**Additional context**
If I force closed the app and reopen, `getToken()` completes. It did not complete when I call `getToken()` directly after calling `deleteInstanceID()`. 

Is this the expected behavior?

firebase_messaging version **5.1.1**
flutter doctor output:
```
[✓] Flutter (Channel stable, v1.7.8+hotfix.4, on Mac OS X 10.14.6 18G95, locale
    en-ID)
    • Flutter version 1.7.8+hotfix.4 at /Users/Dev/flutter
    • Framework revision 20e59316b8 (3 months ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.5.1

[✓] Connected device (1 available)
    • Dev’s iPhone • ios • iOS 12.4.1
```

**Comment by crizant** [comment]
Same issue on version 5.1.6 as well.

**Comment by julianbraasch** [comment]
The issue is also present when ```getToken()``` is called before ```deleteInstanceID()```. In my app, I call the following code when the user logs out.

```dart
final FirebaseMessaging _firebaseMessaging = FirebaseMessaging();
final String token = await _firebaseMessaging.getToken();
_firebaseMessaging.deleteInstanceID();
```

This only works once. If the user signs in and tries to sign out again, nothing happens because the Future does not complete.
I'm also running version 5.1.6

**Comment by hebersousa** [comment]
I have not find any more problems in flutter stable v1.9.1+hotfix.6.

**Comment by julianbraasch** [comment]
Seems to be fixed, I'm currently using version 6.0.3 (it's not the very recent version, I know) and cannot reproduce my problems.

**Comment by skybur** [comment]
Tried to reproduce this and the token is properly returned now. Looks like this is fixed lie julianbraasch said.
